src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,223 @@
+        ) {
+            revert InvalidAssetType();
+        }
+
+        if (
+            deployedTokens[_inputAssetA.erc20Address] == address(0) &&
+            deployedTokens[_outputAssetA.erc20Address] == address(0)
+        ) {
+            revert UnknownAssetA(); // invalid address or pool has not been loaded yet/token not deployed yet  --> this will need to get tested
+        }
+
+        PoolInfo memory selectedPool;
+
+        if (deployedTokens[_inputAssetA.erc20Address] == _outputAssetA.erc20Address) {
+            selectedPool = pools[_inputAssetA.erc20Address];
+
+            uint256 startCurveRewards = ICurveRewards(selectedPool.curveRewards).balanceOf(address(this));
+
+            BOOSTER.deposit(selectedPool.poolId, _totalInputValue, true);
+
+            uint256 endCurveRewards = ICurveRewards(selectedPool.curveRewards).balanceOf(address(this));
+
+            outputValueA = (endCurveRewards - startCurveRewards);
+
+            IRepConvexToken(_outputAssetA.erc20Address).mint(_totalInputValue);
+        } else if (deployedTokens[_outputAssetA.erc20Address] == _inputAssetA.erc20Address) {
+            // withdrawal
+
+            selectedPool = pools[_outputAssetA.erc20Address];
+
+            uint256 startCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            // transfer CONVEX tokens from CrvRewards back to the bridge
+            ICurveRewards(selectedPool.curveRewards).withdraw(_totalInputValue, true); // always claim rewards
+
+            BOOSTER.withdraw(selectedPool.poolId, _totalInputValue);
+
+            uint256 endCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            outputValueA = (endCurveLpTokens - startCurveLpTokens);
+
+            IRepConvexToken(_inputAssetA.erc20Address).burn(_totalInputValue);
+        } else {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        // Pay out subsidy to the rollupBeneficiary
+        SUBSIDY.claimSubsidy(
+            computeCriteria(_inputAssetA, _inputAssetB, _outputAssetA, _outputAssetB, 0),
+            _rollupBeneficiary
+        );
+    }
+
+    /**
+    @notice Loads pool information for all pools supported by Convex Finance.
+    @notice Set allowance for Rollup's Curve LP tokens.
+    @notice Cached. Loads only new pools.
+    */
+    function loadPools() public {
+        uint256 currentPoolLength = BOOSTER.poolLength();
+        if (currentPoolLength != poolLength) {
+            for (uint256 i = poolLength; i < currentPoolLength; i++) {
+                (address curveLpToken, address convexToken, , address curveRewards, , ) = BOOSTER.poolInfo(i);
+                pools[curveLpToken] = PoolInfo(uint96(i), convexToken, curveRewards);
+
+                // deploy token, log token address
+                address deployedToken = address(new RepresentingConvexToken("RepresentingConvexToken", "RCT"));
```

Comment:
Since deploying this contract is expensive it would make sense to optimize it. One way to do it is to use a proxy pattern. Deploy this contract only in constructor and save it's address to implementation immutable variable. Then here use the [clone library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/Clones.sol) to deploy a small proxy which will point to the implementation.

Reply:
Resolved